pkg/apis/middleware/session.go

```diff
@@ -40,10 +40,6 @@ func CreateTokenToSessionFunc(verify VerifyFunc) TokenToSessionFunc {
 			claims.Email = claims.Subject
 		}
 
-		if claims.Verified != nil && !*claims.Verified {
-			return nil, fmt.Errorf("email in id_token (%s) isn't verified", claims.Email)
-		}
-
```

Comment:
This impacts every single provider we have right? Are we sure that this is a safe change to make? Is every provider implementing their own verified check?

I'd need to double check this, but IIRC, this is only used for the extra JWT issuers path right?

Reply:
Thanks @JoelSpeed  for your feedback.

I might be wrong but IIRC this check was coming after `verifyEmail` from `pkg/providers/oidc/verifier.go#124` so it overrules the behaviour I implemented. In my first implementation [(here)](https://github.com/oauth2-proxy/oauth2-proxy/pull/2308/commits/e8c77783964d2a93c0acdd4d352827a569e6981e#diff-84dc7de2b2111fe29c5f145a3396d0e26874cdd70872131494b7e783389e4d4e) at the beginning I just modified this function but that was not very clean implemented.

Maybe you have an idea how to handle this after double checking it?

Reply:
@JoelSpeed ?